app/helpers/ansible_credential_helper/textual_summary.rb

```diff
@@ -16,19 +16,22 @@ def textual_group_options
       options << key
 
       define_singleton_method "textual_#{key}" do
-        h = {:label => _(value[:label]), :title => _(value[:help_text])}
-        h[:value] = if value[:type] == :password && (@record.try(key).present? || @record.options[key].present?)
-                      '●●●●●●●●'
-                    else
-                      @record.try(key) || @record.options[key]
-                    end
-        h
+        {
+          :label => _(value[:label]),
+          :title => _(value[:help_text]),
+          :value => attribute_value(value[:type], key, @record)
+        }
       end
     end
 
     TextualGroup.new(_("Credential Options"), options)
   end
 
+  private def attribute_value(attr_type, key, rec)
+    val = (rec.try(key) || rec.options.try(:[], key)).presence
+    attr_type == :password && val ? '●●●●●●●●' : val
```

Comment:
I assume this is pulled from `API_OPTIONS`, correct?

Reply:
What is the "this"?

If it's the value of `attr_type`, then yes, it comes from the API_OPTIONS.

Reply:
Yes sorry.  I had this review left half finished, and added this comment in a rush.  You are correct, and that is what I meant.  Thanks.